client/src/components/Session.jsx

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import { Link, useHistory } from 'react-router-dom';
+
+export default function Session ({ setTopLevelState }) {
+  const history = useHistory();
+  const [sessionId, setSessionId] = useState('');
+  const [loadingSession, setLoadingSession] = useState(false);
+
+  const handleSessionIdChange = (e) => {
+    setSessionId(e.target.value);
+  };
+
+  const handleJoinSession = async () => {
+    setLoadingSession(true);
+
+    const sessionResponse = await axios.get('/users/getSession');
+    const allGuests = [
+      sessionResponse.data.host,
+      ...sessionResponse.data.guests
+    ];
+    setTopLevelState('guests', allGuests);
+
+    // load restaurant data via restaurant api
+    // In the future, will be a query that only returns a single restaurant by search query
+    // Search query will use session data from the db
+    const restaurantResponse = await axios.get('/users/testzip');
```

Comment:
I may be misunderstanding this, but shouldn't we be getting the restaurant + menu data from the top-level state? What I have in my "Create Session" component right now is this same code - querying the API (mock data for now) to get restaurants based on the user's search query, and when the user selects a restaurant, querying for that restaurant's menu - and then saving the restaurant + menu in the top-level state.

Reply:
I think this will only need to pull the menu for this session from the DB and set the user's top-level state. If a user joins a session, the restaurant will have been chosen already and the menu should be in the DB.